lib/rpmsg/rpmsg_internal.h

```diff
@@ -87,6 +87,33 @@ struct rpmsg_ns_msg {
 	uint32_t flags;
 } METAL_PACKED_END;
 
+/**
+ * rpmsg_initialize_ept - initialize rpmsg endpoint
+ *
+ * Initialize an RPMsg endpoint with a name, source address,
+ * remoteproc address, endpoint callback, and destroy endpoint callback.
+ *
+ * @ept: pointer to rpmsg endpoint
+ * @name: service name associated to the endpoint
+ * @src: local address of the endpoint
+ * @dest: target address of the endpoint
+ * @cb: endpoint callback
+ * @ns_unbind_cb: end point service unbind callback, called when remote ept is
+ *                destroyed.
+ */
+static inline void rpmsg_initialize_ept(struct rpmsg_endpoint *ept,
+					const char *name,
+					uint32_t src, uint32_t dest,
+					rpmsg_ept_cb cb,
+					rpmsg_ns_unbind_cb ns_unbind_cb)
+{
+	strncpy(ept->name, name ? name : "", sizeof(ept->name));
+	ept->addr = src;
+	ept->dest_addr = dest;
+	ept->cb = cb;
+	ept->ns_unbind_cb = ns_unbind_cb;
```

Comment:
can you call rpmsg_register_endpoint here? ept is useless without this call.

Reply:
I saw your proposal in your PR. I hesitated because rpmsg_initialize_ept is now only a helper function to initialize the ept.

I kept it that way to minimize the impact of the code, in case someone complains about using this API. Even though I don't think the API is usable.
So I would prefer to minimize the update in this step and then, when we will remove the API (in at least two versions), we could merge rpmsg_initialize_ept into rpmsg_register_endpoint. 
Do you see any potential issue to do it  in this way ?

@edmooring : any opinion on this point?

Reply:
It's better to make the new API work like rpmsg_init_ept at the beginning, or don't add it at all. It's hard to change the behaviour late without breaking the compatibility.

Reply:
No sure to understand your point as there is no new API. The rpmsg_initialize_ept  is in rpmsg_internal.h not exposed to application. 
So no compatibility issue just a refactoring of some internal functions. 
AT the end,  this is probably more a question of reverting a patch, so both solutions are ok for me , just waiting @edmooring  feedback before deciding.

Reply:
I agree with Arnaud about minimizing the change in this step. Keep the existing behavior and deprecate the API.

Reply:
Ok, let do this in 2 steps, we will optimize the code when we will suppress the API after the deprecated period.